app/models/optimized_image.rb

```diff
@@ -49,7 +49,8 @@ def self.create_for(upload, width, height, opts = {})
     end
 
     # prefer to look up the thumbnail without grabbing any locks
-    thumbnail = find_by(upload_id: upload.id, width: width, height: height)
+    extension = ".#{opts[:format] || upload.extension}"
+    thumbnail = find_by(upload_id: upload.id, width: width, height: height, extension: extension)
```

Comment:
Nitpick: you can omit the variables when the keys have the same name

```suggestion
    thumbnail = find_by(upload_id: upload.id, width:, height:, extension:)
```

Reply:
oh til